### ISSUE TITLE
renderDotContent is not aligned to dot

### ISSUE THREAD

**Issue by ReallyLiri** [indiespirit/react-native-chart-kit#285]
My code:

```
            <LineChart
                data={{labels, datasets}}
                width={Dimensions.get('window').width * 0.8}
                height={220}
                chartConfig={{...}}
                bezier
                fromZero
                style={{padding: 10}}
                renderDotContent={({x, y, index}) => <Text>{y}</Text>}
            />
```

Results in the dot text floating without any correlation to the chart:
![image](https://user-images.githubusercontent.com/16985735/75631466-c350e900-5bfb-11ea-9c01-b7e8ae61613e.png)

Is there any config I'm missing?

**Comment by fung199609** [comment]
you can use absolute to set the position of points:

<Text style={{position: 'absolute', top: y, left: x,}}>{y}</Text>

**Comment by liiri** [comment]
Thanks, it almost works, I'm getting the render method called twice with same input but one is placed on the far left for some reason

![image](https://user-images.githubusercontent.com/53004117/77252886-85c20780-6c5f-11ea-895c-fbb001694f20.png)

**Comment by liiri** [comment]
the second instance just seem to ignore the style, I'm guessing if it was always there but with same style the duplicates would just hide each other

```
renderDotContent={({x, y, index}) => <Text style={{position: 'relative', top: y, left: x, backgroundColor: 'red'}}>{index === 0 ? `(${x}, ${y}, ${index})` : ''}</Text>}
```

![image](https://user-images.githubusercontent.com/53004117/77252968-1bf62d80-6c60-11ea-9908-42679f11094c.png)

**Comment by MaxDagerbratt** [comment]
Im getting the same problem even though i'm using position: 'absolute'.
 
My code:
`width={Dimensions.get('window').width} // from react-native
          height={220}
          yAxisInterval={1}
          fromZero
          renderDotContent={({ x, y }) => <Text key={x + y} style={{ position: 'absolute', top: y, left: x }}>{y}</Text>}`

<img width="218" alt="Screenshot 2020-04-03 at 11 03 51" src="https://user-images.githubusercontent.com/17478366/78343295-eaa02a80-759a-11ea-980e-6ee9dbba8bd2.png">

**Comment by Kevzz81517** [comment]
@fung199609 Facing the same problem, tried position: 'absolute'.

**Comment by hoanglam10499** [comment]
same, have solved? :(

**Comment by Kevzz81517** [comment]
> same, have solved? :(

Nope, @liiri 's observation seems correct, style is not  being applied to the coordinates other than 1st coordinate

**Comment by hoanglam10499** [comment]
I think I solved it
`Before:`
![Screen Shot 2020-05-19 at 08 55 30](https://user-images.githubusercontent.com/37326429/82276376-4be75600-99af-11ea-87d0-c75ae97413a1.png)
`After:`
![Screen Shot 2020-05-19 at 08 55 43](https://user-images.githubusercontent.com/37326429/82276382-4f7add00-99af-11ea-8345-12b973b71ba6.png)

**Comment by ReallyLiri** [comment]
Works great thanks!

![image](https://user-images.githubusercontent.com/16985735/82327634-5f161800-99e7-11ea-8756-400c2b464b2f.png)

**Comment by kunalgandhi28** [comment]
how did you pick the y axis value in renderDotContent?

**Comment by ReallyLiri** [comment]
```
 <LineChart
                data={{labels, datasets}}
                renderDotContent={({x, y, index}) => selectedDotIndex === index && <DotContent key={index} x={x} y={y} value={datasets[0].data[index]}/>}
                ...
```

**Comment by azuxx** [comment]
None of the above worked for me. I am using version 5.6.1 and after many hours of attempts I got the following to work as needed:

```javascript
import {Svg, Text as TextSVG}                                   from 'react-native-svg';

//LineChart
              //...
              bezier
              style={{
                marginLeft: chartMarginLeft,
                paddingTop: 30
              }}
              renderDotContent={({x, y, index}) => {
                return (
                  <View key={index}>
                    <Svg>
                      <TextSVG
                        x={x}
                        y={y - 10}
                        fill="black"
                        fontSize="16"
                        fontWeight="normal"
                        textAnchor="middle">
                        {chartDataMock[index]}
                      </TextSVG>
                    </Svg>
                  </View>
                );
              }}
```
<img width="356" alt="a" src="https://user-images.githubusercontent.com/16132913/97334307-e4e29d00-187c-11eb-935a-3ff2aa4ccf78.png">

**Comment by AnujaVelaskar** [comment]
> how did you pick the y axis value in renderDotContent?

I am facing same problem

**Comment by LhonRafaat** [comment]
> None of the above worked for me. I am using version 5.6.1 and after many hours of attempts I got the following to work as needed:
> 
> ```js
> import {Svg, Text as TextSVG}                                   from 'react-native-svg';
> 
> //LineChart
>               //...
>               bezier
>               style={{
>                 marginLeft: chartMarginLeft,
>                 paddingTop: 30
>               }}
>               renderDotContent={({x, y, index}) => {
>                 return (
>                       <TextSVG
>                         key={index}
>                         x={x}
>                         y={y - 10}
>                         fill="black"
>                         fontSize="16"
>                         fontWeight="normal"
>                         textAnchor="middle">
>                         {chartDataMock[index]}
>                       </TextSVG>
>                 );
>               }}
> ```
> 
> <img alt="a" width="356" src="https://user-images.githubusercontent.com/16132913/97334307-e4e29d00-187c-11eb-935a-3ff2aa4ccf78.png">

this worked for me
